agent/consul/config_endpoint.go

```diff
@@ -367,11 +357,29 @@ func (c *ConfigEntry) ResolveServiceConfig(args *structs.ServiceConfigRequest, r
 				reply.ProxyConfig = mapCopy.(map[string]interface{})
 				reply.MeshGateway = proxyConf.MeshGateway
 				reply.Expose = proxyConf.Expose
+
+				// Extract the global protocol from proxyConf for upstream configs.
+				rawProtocol := proxyConf.Config["protocol"]
+				if rawProtocol != nil {
+					proxyConfGlobalProtocol, ok = rawProtocol.(string)
+					if !ok {
+						return fmt.Errorf("invalid protocol type %T", rawProtocol)
+					}
+				}
 			}
```

Comment:
If `proxyConfGlobalProtocol` is empty, should you initialize it to `tcp` here? Or is it totally fine for the protocol to end up as emptystring?

Reply:
That should be OK. If the proxy-defaults protocol gets overwritten by the higher precedence config like the upstream's service-defaults. If all of them are empty then the protocol key isn't added to the resolved opaque config map (which is the current behavior).